core/types/types.cc

```diff
@@ -1075,15 +1082,15 @@ TypePtr Types::applyTypeArguments(const GlobalState &gs, const CallLocs &locs, u
             if (validBounds) {
                 targs.emplace_back(argType);
             } else {
-                targs.emplace_back(Types::untypedUntracked());
+                targs.emplace_back(Types::untyped(Symbols::Magic_UntypedSource_applyTypeArguments()));
```

Comment:
This is again an error case. `validBounds` is only ever `false` if we reported an error.